### PR TITLE
os/bluestore: optimize _collection_list.

### PR DESCRIPTION
**Pull request by majianpeng**

### CODE REVIEW

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -7213,8 +7213,7 @@ int BlueStore::_collection_list(
   if (!pnext)
     pnext = &static_next;
 
-  if (start == ghobject_t::get_max() ||
-    start.hobj.is_max()) {
+  if (start.is_max()) {
```
Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: this is important because we want to skip out even if ghobject_t::max is false but hobj.max is true.

**Comment by majianpeng** [comment]
update. BTW Whan't mean if ghobject_t::is_max is false but hobj.is_max is true? Thanks!

**Comment by liewegas** [comment]
See 3fa09f02d7b9c16b1b2bd8e8ee09930e61c74a0d. If I remember correctly it was either this, or adding a special case in the caller so that if we have hobject_t max and are composing the ghobject_t we conditoinally do {shard,hobj} or max depending on whether hobj.is_max().  This was simpler (since <shard,max> is still clearly the end of the collection).

**Comment by liewegas** [comment]
Thanks!

**Comment by majianpeng** [comment]
Thanks very much sage!

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-11-18_15:01:48-rados-wip-kefu-testing-2017-11-18-2124-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-20_05:45:10-rados-wip-kefu-testing-2017-11-18-2124-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-11-20_06:28:09-rados-wip-kefu-testing-2017-11-18-2124-distro-basic-mira/

**Pull request merged**